resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/LocalDirsFeatureStep.scala

```diff
@@ -43,33 +44,52 @@ private[spark] class LocalDirsFeatureStep(
     val localDirVolumes = resolvedLocalDirs
       .zipWithIndex
       .map { case (localDir, index) =>
-        new VolumeBuilder()
-          .withName(s"spark-local-dir-${index + 1}")
-          .withNewEmptyDir()
-            .withMedium(if (useLocalDirTmpFs) "Memory" else null)
-          .endEmptyDir()
-          .build()
+        val name = s"spark-local-dir-${index + 1}"
+        findVolume(pod, name) match {
+          case Some(volume) => volume
+          case None =>
+            new VolumeBuilder()
+              .withName(s"spark-local-dir-${index + 1}")
```

Comment:
What is the scope of visibility on volume names "spark-local-dir-1", "spark-local-dir-2" ...
Can those names collide with volumes created by other running jobs?

Reply:
It should be visible for global scope. Since the task distributed should run after setting up executor, so it will not collide with volumes which are built later.

Reply:
volume names in this context are scoped to the pod spec